src/ViewModel/CallToAction.php

```diff
@@ -13,19 +14,32 @@ final class CallToAction implements ViewModel
     use ArrayFromProperties;
 
     private $button;
+    private $dismissible;
     private $id;
     private $image;
     private $needsJs;
     private $text;
 
-    public function __construct(string $id, Picture $image, string $text, Button $button, bool $needsJs = false)
-    {
+    public function __construct(
+        string $id,
+        Picture $image,
+        string $text,
+        Button $button,
+        bool $needsJs = false,
+        DateTimeImmutable $cookieExpires = null
+    ) {
         Assertion::notBlank($id);
         Assertion::notBlank($text);
 
         $this->id = $id;
         if ($needsJs) {
             $this->needsJs = $needsJs;
+
+            if (null !== $cookieExpires) {
+                $this->dismissible = [
+                    'cookieExpires' => $cookieExpires->format(DATE_COOKIE),
```

Comment:
The mustache template accesses `{{cookieExpires}}` rather than traversing `dismissible` first to get to `cookieExpires`, am I missing something?

Reply:
This bit of the mustache template accesses `cookieExpires` from within `dismissible`.
```
      {{#dismissible}}
        data-cookie-name-root="callToAction_"
        {{#cookieExpires}} data-cookie-expires="{{cookieExpires}}"{{/cookieExpires}}
      {{/dismissible}}>
```
(see https://github.com/elifesciences/patterns-php/pull/706/files#diff-7785a3b52eb7ba9a87cbce26c8d0953eR9).

Not sure if this is what you're referring to, or whether there's something else?

Reply:
Yes, I didn't realize the scope changed inside the `dismissible` section; this all works.